crates/bitwarden-ssh/src/import.rs

```diff
@@ -0,0 +1,214 @@
+use ed25519;
+use pkcs8::{der::Decode, pkcs5, DecodePrivateKey, PrivateKeyInfo, SecretDocument};
+use ssh_key::private::{Ed25519Keypair, RsaKeypair};
+
+use crate::{error::SshKeyImportError, SshKey};
+
+const PKCS1_LABEL: &str = "RSA PRIVATE KEY";
+const PKCS8_UNENCRYPTED_LABEL: &str = "PRIVATE KEY";
+const PKCS8_ENCRYPTED_LABEL: &str = "ENCRYPTED PRIVATE KEY";
+const OPENSSH_LABEL: &str = "OPENSSH PRIVATE KEY";
+
+pub fn import_key(
+    encoded_key: String,
+    password: Option<String>,
+) -> Result<SshKey, SshKeyImportError> {
+    let label = pem_rfc7468::decode_label(encoded_key.as_bytes())
+        .map_err(|_| SshKeyImportError::ParsingError)?;
+
+    match label {
+        PKCS1_LABEL => Err(SshKeyImportError::UnsupportedKeyType),
+        PKCS8_UNENCRYPTED_LABEL => import_pkcs8_key(encoded_key, None),
+        PKCS8_ENCRYPTED_LABEL => import_pkcs8_key(
+            encoded_key,
+            Some(password.ok_or(SshKeyImportError::PasswordRequired)?),
+        ),
+        OPENSSH_LABEL => import_openssh_key(encoded_key, password),
+        _ => Err(SshKeyImportError::ParsingError),
```

Comment:
⚠️ Make sure you cover your error cases in your tests!

Reply:
I think this one specifically was covered by `import_non_key_error`, now that we switched to using pem_rfc7468 we need a new test though.